### PR TITLE
[core] Change available_resources and cluster_resources to GiB

### PR DESCRIPTION
**Pull request by ijrsvt**
## Related issue number
`ray.cluster_resources()` and `ray.available_resources()` used to display memory in `50MiB` increments. This is useful for internals of Ray, but is a bit confusing to a user who wants to look at resources. ~The new result does change the returned type of `memory` and `object_store_memory` from a `float` to a `string`.~ The results are still in `float`.

The new result of `ray.cluster_resources()` is:
```
{'CPU': 16,
 'node:IP_ADDR': 1,
 'memory': '4.05',
 'object_store_memory': '1.37'}
```
Before it was:
```
{'CPU': 16,
 'node:IP_ADDR': 1,
 'memory': 81.0,
 'object_store_memory': 27}
```

This PR also cleans up a few other locations that do this conversion to use the new helper method.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: python/ray/ray_constants.py [hunk 1]
```diff
@@ -72,7 +72,7 @@ def env_bool(key, default):
 PLASMA_RESERVABLE_MEMORY_FRACTION = 0.69
 
 
-def round_to_memory_units(memory_bytes, round_up):
+def round_bytes(memory_bytes, round_up):
```
Comment (ericl) [inline python/ray/ray_constants.py:75]: The previous name is more accurate

**Review by ericl — APPROVED** [review]

File: python/ray/ray_constants.py (hunk 1 above)
Comment (simon-mo) [reply python/ray/ray_constants.py:75]: +1

**Review by simon-mo — APPROVED** [review]

File: python/ray/ray_constants.py (hunk 1 above)
Comment (ijrsvt) [reply python/ray/ray_constants.py:75]: But the resulting units are still in bytes?

**Comment by wuisawesome** [comment]
Closing this PR: We promise that `cluster_resources() -> Dict[str, Number]` mapping, but we don't provide semantics guarantees w.r.t the units of internal memory. Therefore (a) we can't change this to a string. (b) we don't _need_ to make the memory more readable (at least for now).

cc @edoakes @ericl @rkooo567

**Comment by ijrsvt** [comment]
@wuisawesome The PR, at this point did not break that. It still keeps it in `Dict[str, Number]` form. While we don't need to change this, it seems worthwhile to clean up this API for the 1.0 release.

**Pull request closed without merging**